api/storage/elasticsearch/elasticsearch.go

```diff
@@ -327,6 +336,63 @@ func (sm *StorageManager) GetResources(resourceType string, executionID string,
 	return resources, nil
 }
 
+// GetResourceTrends return resource data
+func (sm *StorageManager) GetResourceTrends(resourceType string, filters map[string]string) ([]storage.ExecutionCost, error) {
+	// Per resource trends, filters should take care of granularity (per resource: Data.ResourceID, Data.Region, Data.Metric -> Data.PricePerMonth)
+	var resources []storage.ExecutionCost
+	var mustNotQuery []elastic.Query
+
+	// Must
+	mustQuery := sm.getDynamicMatchQuery(filters, "and")
+	mustQuery = append(mustQuery, elastic.NewMatchQuery("ResourceName", resourceType).Operator("and"))
+
+	// Unsupported Types - MustNot
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("EventType", "service_status").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_iam_users").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_elastic_ip").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_lambda").Operator("and"))
+	mustNotQuery = append(mustNotQuery, elastic.NewMatchQuery("ResourceName", "aws_ec2_volume").Operator("and"))
+
+	queryBuilder := elastic.NewBoolQuery().MustNot(mustNotQuery...).Must(mustQuery...)
+	searchResult, err := sm.client.Search().
+		Query(queryBuilder).
+		Pretty(true).
+		Size(100).
```

Comment:
this is potential bug, your elasticsearch query will limit the result to 100. 
First, you need to execute the quest with size `0` then you will get the `TotalHits` (number of records) then you need to execute the query with the total hits response value into the `Size` query. you can see an example in `GetExecutionTags` function

Reply:
Why is it a bug?
I don't want to have more than 100 results, fixed. TBH, I want it to be much lower, I wanted to see how much actions we have first.
Take into account I'm not working on the hits, this query aggregates the data it finds into different buckets and I'm iterating on top of them.